## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 513836`
Commit: [`04177d7f77a0c1bb5d4dee6626ddc5197b8a3525`](https://github.com/NixOS/nixpkgs/commit/04177d7f77a0c1bb5d4dee6626ddc5197b8a3525) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/04177d7f77a0c1bb5d4dee6626ddc5197b8a3525..pull/513836/head))
Merge: [`40b16bc3b8417c5289271d1cc3c9ecd8362d76ef`](https://github.com/NixOS/nixpkgs/commit/40b16bc3b8417c5289271d1cc3c9ecd8362d76ef)

Logs: https://github.com/drawbu/nixpkgs-review-gha/actions/runs/24971697450


---
### `x86_64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>opensaml-cpp</li>
    <li>shibboleth-sp</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>autenticacao-gov-pt-bin</li>
    <li>log4shib</li>
    <li>xml-security-c</li>
    <li>xml-tooling-c</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>opensaml-cpp</summary>
<pre>checking dynamic linker characteristics... (cached) GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking for an ANSI C-conforming const... yes
checking for size_t... yes
checking for dirent.h that defines DIR... yes
checking for library containing opendir... none required
checking for strchr... yes
checking for strdup... yes
checking for strstr... yes
checking how to run the C preprocessor... gcc -E
checking for egrep -e... (cached) /nix/store/wf7lr2hf43546jc5kwqh3dbxnpcnw1mn-gnugrep-3.12/bin/grep -E
checking whether gcc is Clang... no
checking whether pthreads work with -pthread... yes
checking for joinable pthread attribute... PTHREAD_CREATE_JOINABLE
checking whether more special flags are required for pthreads... no
checking for PTHREAD_PRIO_INHERIT... yes
checking whether the compiler implements namespaces... yes
checking whether the compiler supports Standard Template Library... yes
checking for Boost headers version &gt;= 0.0.0... no
configure: error: cannot find Boost headers version &gt;= 0.0.0</pre>
</details>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>opensaml-cpp</li>
    <li>shibboleth-sp</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>autenticacao-gov-pt-bin</li>
    <li>log4shib</li>
    <li>xml-security-c</li>
    <li>xml-tooling-c</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>opensaml-cpp</summary>
<pre>checking dynamic linker characteristics... (cached) GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking for an ANSI C-conforming const... yes
checking for size_t... yes
checking for dirent.h that defines DIR... yes
checking for library containing opendir... none required
checking for strchr... yes
checking for strdup... yes
checking for strstr... yes
checking how to run the C preprocessor... gcc -E
checking for egrep -e... (cached) /nix/store/g7mn7yzwavbs72dv6mcganlkqjj1bq9w-gnugrep-3.12/bin/grep -E
checking whether gcc is Clang... no
checking whether pthreads work with -pthread... yes
checking for joinable pthread attribute... PTHREAD_CREATE_JOINABLE
checking whether more special flags are required for pthreads... no
checking for PTHREAD_PRIO_INHERIT... yes
checking whether the compiler implements namespaces... yes
checking whether the compiler supports Standard Template Library... yes
checking for Boost headers version &gt;= 0.0.0... no
configure: error: cannot find Boost headers version &gt;= 0.0.0</pre>
</details>
</details>

---
### `x86_64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>opensaml-cpp</li>
    <li>shibboleth-sp</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>log4shib</li>
    <li>xml-security-c</li>
    <li>xml-tooling-c</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>opensaml-cpp</summary>
<pre>checking dynamic linker characteristics... darwin24.6.0 dyld
checking how to hardcode library paths into programs... immediate
checking for an ANSI C-conforming const... yes
checking for size_t... yes
checking for dirent.h that defines DIR... yes
checking for library containing opendir... none required
checking for strchr... yes
checking for strdup... yes
checking for strstr... yes
checking how to run the C preprocessor... clang -std=gnu23 -std=gnu23 -E
checking for egrep -e... (cached) /nix/store/ggcs7dhvjza0pl1hnwg7h4gjx4brgnm1-gnugrep-3.12/bin/grep -E
checking whether clang -std=gnu23 -std=gnu23 is Clang... yes
checking whether Clang needs flag to prevent &quot;argument unused&quot; warning when linking with -pthread... no
checking for joinable pthread attribute... PTHREAD_CREATE_JOINABLE
checking whether more special flags are required for pthreads... no
checking for PTHREAD_PRIO_INHERIT... yes
checking whether the compiler implements namespaces... yes
checking whether the compiler supports Standard Template Library... yes
checking for Boost headers version &gt;= 0.0.0... no
configure: error: cannot find Boost headers version &gt;= 0.0.0</pre>
</details>
</details>

---
### `aarch64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>opensaml-cpp</li>
    <li>shibboleth-sp</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>log4shib</li>
    <li>xml-security-c</li>
    <li>xml-tooling-c</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>opensaml-cpp</summary>
<pre>checking dynamic linker characteristics... darwin24.6.0 dyld
checking how to hardcode library paths into programs... immediate
checking for an ANSI C-conforming const... yes
checking for size_t... yes
checking for dirent.h that defines DIR... yes
checking for library containing opendir... none required
checking for strchr... yes
checking for strdup... yes
checking for strstr... yes
checking how to run the C preprocessor... clang -std=gnu23 -std=gnu23 -E
checking for egrep -e... (cached) /nix/store/sk1la8r8sa1rlb8m9kwy5hi29f58sjd9-gnugrep-3.12/bin/grep -E
checking whether clang -std=gnu23 -std=gnu23 is Clang... yes
checking whether Clang needs flag to prevent &quot;argument unused&quot; warning when linking with -pthread... no
checking for joinable pthread attribute... PTHREAD_CREATE_JOINABLE
checking whether more special flags are required for pthreads... no
checking for PTHREAD_PRIO_INHERIT... yes
checking whether the compiler implements namespaces... yes
checking whether the compiler supports Standard Template Library... yes
checking for Boost headers version &gt;= 0.0.0... no
configure: error: cannot find Boost headers version &gt;= 0.0.0</pre>
</details>
</details>

--- a/pkgs/by-name/lo/log4shib/package.nix
+++ b/pkgs/by-name/lo/log4shib/package.nix
@@ -3,6 +3,7 @@
   stdenv,
   fetchFromCodeberg,
   autoreconfHook,
+  unstableGitUpdater,
 }:
 
 stdenv.mkDerivation (finalAttrs: {
@@ -19,6 +20,8 @@ stdenv.mkDerivation (finalAttrs: {
   nativeBuildInputs = [ autoreconfHook ];
 
   env.CXXFLAGS = "-std=c++11";
+
+  passthru.updateScript = unstableGitUpdater { };
 
   meta = {
     description = "Forked version of log4cpp that has been created for the Shibboleth project";

--- a/pkgs/by-name/lo/log4shib/package.nix
+++ b/pkgs/by-name/lo/log4shib/package.nix
@@ -26,7 +26,7 @@ stdenv.mkDerivation (finalAttrs: {
   meta = {
     description = "Forked version of log4cpp that has been created for the Shibboleth project";
     mainProgram = "log4shib-config";
-    maintainers = [ ];
+    maintainers = with lib.maintainers; [ drawbu ];
     license = lib.licenses.lgpl21;
     homepage = "http://log4cpp.sf.net";
   };

--- a/pkgs/by-name/op/opensaml-cpp/package.nix
+++ b/pkgs/by-name/op/opensaml-cpp/package.nix
@@ -53,6 +53,6 @@ stdenv.mkDerivation (finalAttrs: {
     mainProgram = "samlsign";
     platforms = lib.platforms.unix;
     license = lib.licenses.asl20;
-    maintainers = [ ];
+    maintainers = with lib.maintainers; [ drawbu ];
   };
 })

--- a/pkgs/by-name/op/opensaml-cpp/package.nix
+++ b/pkgs/by-name/op/opensaml-cpp/package.nix
@@ -11,6 +11,7 @@
   xml-security-c,
   xml-tooling-c,
   zlib,
+  unstableGitUpdater,
 }:
 
 stdenv.mkDerivation (finalAttrs: {
@@ -43,6 +44,8 @@ stdenv.mkDerivation (finalAttrs: {
   env.NIX_CFLAGS_COMPILE = lib.optionalString (!stdenv.hostPlatform.isDarwin) "-std=c++14";
 
   enableParallelBuilding = true;
+
+  passthru.updateScript = unstableGitUpdater { };
 
   meta = {
     homepage = "https://shibboleth.net/products/opensaml-cpp.html";

--- a/pkgs/by-name/op/opensaml-cpp/package.nix
+++ b/pkgs/by-name/op/opensaml-cpp/package.nix
@@ -39,7 +39,10 @@ stdenv.mkDerivation (finalAttrs: {
     pkg-config
   ];
 
-  configureFlags = [ "--with-xmltooling=${xml-tooling-c}" ];
+  configureFlags = [
+    "--with-xmltooling=${xml-tooling-c}"
+    "--with-boost=${boost.dev}"
+  ];
 
   env.NIX_CFLAGS_COMPILE = lib.optionalString (!stdenv.hostPlatform.isDarwin) "-std=c++14";
 

--- a/pkgs/by-name/sh/shibboleth-sp/package.nix
+++ b/pkgs/by-name/sh/shibboleth-sp/package.nix
@@ -13,6 +13,7 @@
   xercesc,
   xml-security-c,
   xml-tooling-c,
+  unstableGitUpdater,
 }:
 
 stdenv.mkDerivation (finalAttrs: {
@@ -71,6 +72,8 @@ stdenv.mkDerivation (finalAttrs: {
   ];
 
   enableParallelBuilding = true;
+
+  passthru.updateScript = unstableGitUpdater { };
 
   meta = {
     homepage = "https://shibboleth.net/products/service-provider.html";

--- a/pkgs/by-name/sh/shibboleth-sp/package.nix
+++ b/pkgs/by-name/sh/shibboleth-sp/package.nix
@@ -80,6 +80,6 @@ stdenv.mkDerivation (finalAttrs: {
     description = "Enables SSO and Federation web applications written with any programming language or framework";
     platforms = lib.platforms.unix;
     license = lib.licenses.asl20;
-    maintainers = [ ];
+    maintainers = with lib.maintainers; [ drawbu ];
   };
 })

--- a/pkgs/by-name/xm/xml-security-c/package.nix
+++ b/pkgs/by-name/xm/xml-security-c/package.nix
@@ -7,6 +7,7 @@
   xalanc,
   xercesc,
   openssl,
+  unstableGitUpdater,
 }:
 
 stdenv.mkDerivation (finalAttrs: {
@@ -36,6 +37,8 @@ stdenv.mkDerivation (finalAttrs: {
     xercesc
     openssl
   ];
+
+  passthru.updateScript = unstableGitUpdater { };
 
   meta = {
     homepage = "https://shibboleth.atlassian.net/wiki/spaces/DEV/pages/3726671873/Santuario";

--- a/pkgs/by-name/xm/xml-security-c/package.nix
+++ b/pkgs/by-name/xm/xml-security-c/package.nix
@@ -45,5 +45,6 @@ stdenv.mkDerivation (finalAttrs: {
     description = "C++ Implementation of W3C security standards for XML";
     license = lib.licenses.asl20;
     platforms = lib.platforms.unix;
+    maintainers = with lib.maintainers; [ drawbu ];
   };
 })

--- a/pkgs/by-name/xm/xml-tooling-c/package.nix
+++ b/pkgs/by-name/xm/xml-tooling-c/package.nix
@@ -10,6 +10,7 @@
   log4shib,
   xercesc,
   xml-security-c,
+  unstableGitUpdater,
 }:
 
 stdenv.mkDerivation (finalAttrs: {
@@ -44,6 +45,8 @@ stdenv.mkDerivation (finalAttrs: {
   env.NIX_CFLAGS_COMPILE = lib.optionalString (!stdenv.hostPlatform.isDarwin) "-std=c++14";
 
   enableParallelBuilding = true;
+
+  passthru.updateScript = unstableGitUpdater { };
 
   meta = {
     description = "Low-level library that provides a high level interface to XML processing for OpenSAML 2";

